src/app/core/services/scheduling.service.ts

```diff
@@ -226,33 +223,35 @@ export class SchedulingService {
     return this.storage.push(StorageKeys.SCHEDULE_TASKS_COMPLETED, task)
   }
 
-  updateScheduleWithCompletedTasks(schedule) {
+  updateScheduleWithCompletedTasks(schedule): Task[] {
     // NOTE: If utcOffsetPrev exists, timezone has changed
     if (this.utcOffsetPrev != null) {
-      this.storage.remove(StorageKeys.SCHEDULE_TASKS_COMPLETED).then(() => {
-        const currentMidnight = new Date().setHours(0, 0, 0, 0)
-        const prevMidnight =
-          new Date().setUTCHours(0, 0, 0, 0) + this.utcOffsetPrev * 60000
-        this.completedTasks.map(d => {
-          const index = schedule.findIndex(
-            s =>
-              s.timestamp - currentMidnight == d.timestamp - prevMidnight &&
-              s.name == d.name
-          )
-          if (index > -1) {
-            schedule[index].completed = true
-            this.addToCompletedTasks(schedule[index])
-          }
+      this.storage
+        .remove(StorageKeys.SCHEDULE_TASKS_COMPLETED)
+        .then(() => {
+          const currentMidnight = new Date().setHours(0, 0, 0, 0)
+          const prevMidnight =
+            new Date().setUTCHours(0, 0, 0, 0) + this.utcOffsetPrev * 60000
```

Comment:
You can use the time functions created before to get the milliseconds for min.

Reply:
Ok I'll change this.